x-pack/plugins/fleet/server/services/package_policy.ts

```diff
@@ -955,6 +962,77 @@ class PackagePolicyService implements PackagePolicyServiceInterface {
   }
 }
 
+export class PackagePolicyServiceImpl
+  extends PackagePolicyClientImpl
+  implements PackagePolicyService
+{
+  public asScoped(request: KibanaRequest): PackagePolicyClient {
+    const preflightCheck = async (fleetAuthzConfig: FleetAuthzRouteConfig) => {
+      const authz = await getAuthzFromRequest(request);
+      if (!hasRequiredFleetAuthzPrivilege(authz, fleetAuthzConfig)) {
+        throw new FleetUnauthorizedError('Not authorized to this action on integration policies');
+      }
+    };
+    return new PackagePolicyClientWithAuthz(preflightCheck);
+  }
+
+  public get asInternalUser() {
+    return new PackagePolicyClientWithAuthz();
+  }
+}
+
+class PackagePolicyClientWithAuthz extends PackagePolicyClientImpl {
+  constructor(
+    private readonly preflightCheck?: (
+      fleetAuthzConfig: FleetAuthzRouteConfig
+    ) => void | Promise<void>
+  ) {
+    super();
+  }
+
+  #runPreflight = async (fleetAuthzConfig: FleetAuthzRouteConfig) => {
+    if (this.preflightCheck) {
+      return await this.preflightCheck(fleetAuthzConfig);
+    }
+  };
+
+  async create(
+    soClient: SavedObjectsClientContract,
+    esClient: ElasticsearchClient,
+    packagePolicy: NewPackagePolicy,
+    options?: {
+      spaceId?: string;
+      id?: string;
+      user?: AuthenticatedUser;
+      bumpRevision?: boolean;
+      force?: boolean;
+      skipEnsureInstalled?: boolean;
+      skipUniqueNameVerification?: boolean;
+      overwrite?: boolean;
+    }
+  ): Promise<PackagePolicy> {
+    try {
+      await this.#runPreflight({
+        fleetAuthz: {
+          integrations: { writeIntegrationPolicies: true },
+        },
+      });
+    } catch (error) {
+      if (packagePolicy.package?.name === 'endpoint') {
+        await this.#runPreflight({
+          fleetAuthz: {
+            integrations: { writeEndpointIntegration: true },
```

Comment:
the backend check allows `POST /package_policies` call with only endpoint package if user has `writeEndpointIntegration` privilege
<img width="1768" alt="image" src="https://user-images.githubusercontent.com/90178898/165949514-410a56bd-844c-4a22-ad7f-c071906a9284.png">
<img width="1775" alt="image" src="https://user-images.githubusercontent.com/90178898/165949692-992750d7-795d-4c09-842e-ad1cbd793252.png">